src/game/logic/object/weapon.h

```diff
@@ -116,14 +141,154 @@ class WeaponBonusSet : public MemoryPoolObject
     WeaponBonus m_bonus[WEAPONBONUSCONDITION_COUNT];
 };
 
-class WeaponTemplate;
+struct HistoricWeaponDamageInfo
+{
+    unsigned int frame;
+    Coord3D location;
+    HistoricWeaponDamageInfo(unsigned int f, Coord3D loc) : frame(f), location(loc) {}
+};
+
+class WeaponTemplate : public MemoryPoolObject
+{
+    IMPLEMENT_POOL(WeaponTemplate);
+
+public:
+    WeaponTemplate();
+
+    virtual ~WeaponTemplate() override;
+    WeaponTemplate &operator=(const WeaponTemplate &that);
+
+    float Get_Min_Target_Pitch() const { return m_minTargetPitch; }
+    float Get_Max_Target_Pitch() const { return m_maxTargetPitch; }
+    int Get_Shots_Per_Barrel() const { return m_shotsPerBarrel; }
+    unsigned int Get_Suspend_FX_Delay() const { return m_suspendFXDelay; }
+    Utf8String Get_Name() const { return m_name; }
```

Comment:
Does not return const & ?

Reply:
No, definitely returns AsciiString in the original.